### PR TITLE
Jc/command pconvert

### PR DESCRIPTION
**Pull request by joao-conde**
| - | - |
| --- | --- |
| Decisions | Add initial Rust project files and `convert` command. Make use of package https://github.com/image-rs/image for image processing |

### CODE REVIEW

**Comment by joamag** [comment]
@joao-conde please do not forget to assing me as the reviewer and assignee otherwise I won't be notified.

File: Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+[package]
+name = "pconvert-rust"
+version = "0.1.0"
+authors = ["João Dias Conde Azevedo"]
```
Comment (joamag) [inline Cargo.toml:4]: Platforme International instead

File: Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+edition = "2018"
+
+# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
```
Comment (joamag) [inline Cargo.toml:7]: remove this comment

File: README.md [hunk 3]
```diff
@@ -1,3 +1,27 @@
 # P(NG)Convert Rust
 
 The [Rust](https://www.rust-lang.org) version of the famous [P(NG)Convert](https://github.com/hivesolutions/pconvert) from Hive Solutions.
+
+
```
Comment (joamag) [inline README.md:5]: remove extra EOL

File: README.md [hunk 4]
```diff
@@ -1,3 +1,27 @@
...
+## Compiling & Executing
+
+Compile with cargo
+```
```
Comment (joamag) [inline README.md:9]: add the bash reference ```bash

**Review by joamag — CHANGES_REQUESTED** [review]

**Comment by joao-conde** [comment]
@joamag I have fixed the issues you pointed out.

Also:
> @joao-conde please do not forget to assign me as the reviewer and assignee otherwise I won't be notified.

I won't forget again :+1:

File: src/lib.rs [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+    let file_out = match args.next() {
+        Some(name) => name,
+        None => {
+            println!("Missing output path.\nUsage: pconvert convert <file_in> <file_out>");
+            process::exit(0);
+        }
+    };
+
+    //load PNG
```
Comment (joamag) [inline src/lib.rs:22]: remove comment

File: src/lib.rs [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+        _ => {
+            eprintln!("ERROR: Input file given must be a PNG with RGBA components per pixel");
+            process::exit(-1);
+        }
+    };
+
+    //turn the image blueish: "sets red value to 0 and green value to the blue one (blue filter)"
+    img.pixels_mut().for_each(|x| apply_blue_filter(x));
+
+    //save modified PNG
```
Comment (joamag) [inline src/lib.rs:34]: remove comment

File: src/lib.rs [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+    //load PNG
+    let mut img = match open(file_in).expect("Failed to open input file") {
+        DynamicImage::ImageRgba8(img) => img,
+        _ => {
+            eprintln!("ERROR: Input file given must be a PNG with RGBA components per pixel");
+            process::exit(-1);
+        }
+    };
+
+    //turn the image blueish: "sets red value to 0 and green value to the blue one (blue filter)"
```
Comment (joamag) [inline src/lib.rs:31]: `// turns the image`

File: src/main.rs [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+
+fn print_usage() {
+    println!(
+        "Usage: pconvert-rust <command> [args...]\nwhere command can be one of the following: compose, convert, benchmark, opencl, version"
+    );
+}
+
+fn main() {
+    let mut args = env::args();
+    args.next(); // skip program name
```
Comment (joamag) [inline src/main.rs:11]: move the comment up

**Review by joamag — CHANGES_REQUESTED** [review]

**Review by joamag — APPROVED** [review]

**Pull request merged**